OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -879,9 +867,23 @@ function Test-TargetResource
   {
     $currentValue = $currentResource.Item($key)
     $requestedValue = $params.Item($key)
-    if ($key -eq "OctopusAdminPassword")
+    if ($key -eq "OctopusAdminCredential") 
     {
-      if ((Get-DecryptedSecureString $currentValue) -ne $requestedValue)
+      if($null -ne $currentValue)
+      {
+        $currentUsername = $currentValue.GetNetworkCredential().UserName
```

Comment:
Is there a case here where the value in the installstate is a different (ie old) format?

Reply:
... possibly, yes. But given it's a breaking change, I was going to recommend in the release notes that Installations managed by 2.x may cause errors if you flip to 3.x and vice versa, rather than trying to handle every case.

Reply:
Hmmm. Understandable, but if we put the pain on the end user, what'd be the workaround?
I'm inclined to attempt to handle both cases.

Reply:
OK, I'll look at that. It might actually still be using the same installstate as before, but I'll see if I can test that.

Reply:
At the moment, the installstate file has a plaintext password in it (for both 2.0 and 3.0 versions). I can make 3.0 have a securestring or encrypted string instead, but that introduces another breaking change. 

If we want to get rid of the PSAvoidUsingConvertToSecureStringWithPlainText warning, that plain text password has to go.

Reply:
(as discussed in slack) it's not stored in plaintext...
_just wanted to clarify if anyone is watching along at home_